x/fungible/keeper/zevm_msg_passing.go

```diff
@@ -0,0 +1,76 @@
+package keeper
+
+import (
+	"fmt"
+	"math/big"
+
+	"cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	evmtypes "github.com/evmos/ethermint/x/evm/types"
+	"github.com/zeta-chain/zetacore/x/fungible/types"
+)
+
+func (k Keeper) ZEVMDepositAndCallContract(ctx sdk.Context,
+	sender ethcommon.Address,
+	to ethcommon.Address,
+	inboundSenderChainID int64,
+	inboundAmount *big.Int,
+	data []byte,
+	indexBytes [32]byte) (*evmtypes.MsgEthereumTxResponse, error) {
+	acc := k.evmKeeper.GetAccount(ctx, to)
+	if acc == nil {
+		return nil, errors.Wrap(types.ErrAccountNotFound, fmt.Sprintf("address: %s", to.String()))
+	}
+	if !acc.IsContract() {
+		err := k.DepositCoinZeta(ctx, to, inboundAmount)
+		if err != nil {
+			return nil, err
+		}
+		return nil, nil
+	}
+	return k.ZevmOnReceive(ctx, sender.Bytes(), to, big.NewInt(inboundSenderChainID), inboundAmount, data, indexBytes)
+
+}
+func (k Keeper) ZevmOnReceive(ctx sdk.Context,
```

Comment:
Inconsistent with other method prefixed with `ZEVM`, we should use `zevm` everywhere
If it is an internal function, should be `zevm`